packages/grafana-ui/src/components/ColorPicker/ColorPickerInput.tsx

```diff
@@ -48,6 +48,13 @@ export const ColorPickerInput = forwardRef<HTMLInputElement, ColorPickerInputPro
       [currentColor]
     );
 
+    const handleBlur = (evt: FocusEvent<HTMLInputElement>) => {
+      // Unless the user clicked inside the color picker, close it on blur
+      if (!Boolean(evt.relatedTarget?.className?.split(' ').some((c) => /react-colorful/.test(c)))) {
```

Comment:
```suggestion
      if (!evt.relatedTarget?.className.includes('react-colorful')) {
```
I think this can be simplified a bit.

Reply:
we never want to rely on class names since they could change at any moment, e.g. if we update the library. think a more robust solution should rely on something we control:
```tsx
      const isClickInPopover = document.querySelector('[data-testid="color-popover"]')?.contains(evt.relatedTarget);
      if (!isClickInPopover) {
        setIsOpen(false);
      }
```
or maybe we could pass an explicit `id` instead of reusing the `data-testid`. it doesn't really matter 🤷

Reply:
+1 - would rather be passing in an explicit ID/ref than relying on a class name remaining stable.

Reply:
Hey folks, thanks for the feedback and good points. I updated to use an `id` for the targeting method. Let me know what you think.